### ISSUE TITLE
Title in timeline crashes program

### ISSUE THREAD

**Issue by AGI-THB** [olive-editor/olive#638]
**Olive version:**  789b751
**Source:** Appimage
**Operating system:** Linux Mint
**CPU:**  Intel i5-4200u
**RAM:** 16GB
**GPU:** IGPU

### Detailed Description

Add a Video file to the timeline
Then add a title
Then select the titlebar in  the timeline
then click the textbox in the effects panel
then skimm over the titlebar using the red arrow (do it back and forth fast)
program crashes 

### Output Log
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/blackhat/Desktop/Coursera/Georgia_Tech/Electronic_Analysis/1_Direct_Current_Analysis/Olive-789b751-x86_64.AppImage...(no debugging symbols found)...done.
(gdb) run
Starting program: /home/blackhat/Desktop/Coursera/Georgia_Tech/Electronic_Analysis/1_Direct_Current_Analysis/Olive-789b751-x86_64.AppImage 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
process 8553 is executing new program: /tmp/.mount_Olive-qMwbUl/usr/bin/olive-editor
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffd7a67700 (LWP 8562)]
2019-03-16T23:09:49 [WARNING] QApplication: invalid style override passed, ignoring it.
2019-03-16T23:09:49 [INFO] Initializing custom cursors
2019-03-16T23:09:49 [INFO] Finished initializing custom cursors
2019-03-16T23:09:49 [INFO] Initializing icons
2019-03-16T23:09:49 [INFO] Finished initializing icons
[New Thread 0x7fffcffff700 (LWP 8563)]
[New Thread 0x7fffcf7fe700 (LWP 8564)]
[New Thread 0x7fffceffd700 (LWP 8565)]
[New Thread 0x7fffce7fc700 (LWP 8566)]
2019-03-16T23:09:49 [INFO] Initializing effects...
[New Thread 0x7fffb5d5f700 (LWP 8567)]
[New Thread 0x7fffb555e700 (LWP 8568)]
2019-03-16T23:09:52 [INFO] Finished initializing effects
[Thread 0x7fffce7fc700 (LWP 8566) exited]
2019-03-16T23:09:52 [WARNING] QFileInfo::absolutePath: Constructed with empty filename
2019-03-16T23:09:52 [INFO] Matched "Desktop/OpenHPI/Datenbanken/01/01_Einfuehrung.mp4" with absolute path
2019-03-16T23:09:52 [WARNING] Tried to close an effect that was already closed
[New Thread 0x7fffce7fc700 (LWP 8569)]
[Thread 0x7fffb555e700 (LWP 8568) exited]
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/home/blackhat/Desktop/OpenHPI/Datenbanken/01/01_Einfuehrung.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp42mp41isomavc1
    creation_time   : 2014-05-06T09:10:34.000000Z
  Duration: 00:34:16.53, start: 0.000000, bitrate: 616 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1280x720 [SAR 1:1 DAR 16:9], 452 kb/s, 25 fps, 25 tbr, 25 tbn, 50 tbc (default)
    Metadata:
      creation_time   : 2014-05-06T09:10:34.000000Z
      handler_name    : L-SMASH Video Handler
      encoder         : AVC Coding
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 160 kb/s (default)
    Metadata:
      creation_time   : 2014-05-06T09:10:34.000000Z
      handler_name    : L-SMASH Audio Handler
2019-03-16T23:09:53 [WARNING] QObject::connect: Cannot queue arguments of type 'QVector<int>'
(Make sure 'QVector<int>' is registered using qRegisterMetaType().)
[Thread 0x7fffce7fc700 (LWP 8569) exited]
[New Thread 0x7fffce7fc700 (LWP 8570)]
[New Thread 0x7fffb555e700 (LWP 8571)]
[New Thread 0x7fffa895b700 (LWP 8572)]
[New Thread 0x7fffb095b700 (LWP 8573)]
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/home/blackhat/Desktop/OpenHPI/Datenbanken/01/01_Einfuehrung.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp42mp41isomavc1
    creation_time   : 2014-05-06T09:10:34.000000Z
  Duration: 00:34:16.53, start: 0.000000, bitrate: 616 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1280x720 [SAR 1:1 DAR 16:9], 452 kb/s, 25 fps, 25 tbr, 25 tbn, 50 tbc (default)
    Metadata:
      creation_time   : 2014-05-06T09:10:34.000000Z
      handler_name    : L-SMASH Video Handler
      encoder         : AVC Coding
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 160 kb/s (default)
    Metadata:
      creation_time   : 2014-05-06T09:10:34.000000Z
      handler_name    : L-SMASH Audio Handler
[New Thread 0x7fffabfff700 (LWP 8574)]
[New Thread 0x7fffab7fe700 (LWP 8575)]
[New Thread 0x7fffaaffd700 (LWP 8576)]
[New Thread 0x7fffaa7fc700 (LWP 8577)]
2019-03-16T23:09:55 [INFO] Clip opened on track 0 (took 61 ms)
[New Thread 0x7fffa0db0700 (LWP 8578)]
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/home/blackhat/Desktop/OpenHPI/Datenbanken/01/01_Einfuehrung.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp42mp41isomavc1
    creation_time   : 2014-05-06T09:10:34.000000Z
  Duration: 00:34:16.53, start: 0.000000, bitrate: 616 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1280x720 [SAR 1:1 DAR 16:9], 452 kb/s, 25 fps, 25 tbr, 25 tbn, 50 tbc (default)
    Metadata:
      creation_time   : 2014-05-06T09:10:34.000000Z
      handler_name    : L-SMASH Video Handler
      encoder         : AVC Coding
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 160 kb/s (default)
    Metadata:
      creation_time   : 2014-05-06T09:10:34.000000Z
      handler_name    : L-SMASH Audio Handler
[New Thread 0x7fff97fff700 (LWP 8579)]
[New Thread 0x7fff977fe700 (LWP 8580)]
[New Thread 0x7fff96ffd700 (LWP 8581)]
[New Thread 0x7fff967fc700 (LWP 8582)]
[New Thread 0x7fff95ffb700 (LWP 8583)]
[New Thread 0x7fff957fa700 (LWP 8584)]
[New Thread 0x7fff94ff9700 (LWP 8585)]
[New Thread 0x7fff947f8700 (LWP 8586)]
[New Thread 0x7fff93ff7700 (LWP 8587)]
2019-03-16T23:09:55 [INFO] Clip opened on track -1 (took 49 ms)
2019-03-16T23:10:01 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:02 [WARNING] QTextCursor::setPosition: Position '12' out of range
2019-03-16T23:10:02 [WARNING] QTextCursor::setPosition: Position '12' out of range
2019-03-16T23:10:03 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:03 [WARNING] QTextCursor::setPosition: Position '11' out of range
2019-03-16T23:10:03 [WARNING] QTextCursor::setPosition: Position '11' out of range
2019-03-16T23:10:03 [WARNING] QTextCursor::setPosition: Position '11' out of range
2019-03-16T23:10:03 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:03 [WARNING] QTextCursor::setPosition: Position '10' out of range
2019-03-16T23:10:03 [WARNING] QTextCursor::setPosition: Position '10' out of range
2019-03-16T23:10:03 [WARNING] QTextCursor::setPosition: Position '10' out of range
2019-03-16T23:10:04 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '9' out of range
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '9' out of range
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '9' out of range
2019-03-16T23:10:04 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '8' out of range
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '8' out of range
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '8' out of range
2019-03-16T23:10:04 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '7' out of range
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '7' out of range
2019-03-16T23:10:04 [WARNING] QTextCursor::setPosition: Position '7' out of range
2019-03-16T23:10:05 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '8' out of range
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '8' out of range
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '8' out of range
2019-03-16T23:10:05 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '9' out of range
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '9' out of range
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '9' out of range
2019-03-16T23:10:05 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '10' out of range
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '10' out of range
2019-03-16T23:10:05 [WARNING] QTextCursor::setPosition: Position '10' out of range
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to start a closed effect - opening
2019-03-16T23:10:08 [WARNING] Invalid values passed to get_buffer_offset_from_frame 643 < 661
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Invalid values passed to get_buffer_offset_from_frame 670 < 690
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Invalid values passed to get_buffer_offset_from_frame 725 < 727
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to start a closed effect - opening
2019-03-16T23:10:08 [WARNING] Tried to start a closed effect - opening
2019-03-16T23:10:08 [WARNING] Texture is not valid in the current context.
Texture has not been destroyed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:10:08 [WARNING] Tried to close an effect that was already closed

Thread 4 "RenderThread" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffcf7fe700 (LWP 8564)]
0x00007ffff4c4889b in QOpenGLTexture::generateMipMaps() () from /tmp/.mount_Olive-qMwbUl/usr/bin/../lib/libQt5Gui.so.5

(gdb) bt
#0  0x00007ffff4c4889b in QOpenGLTexture::generateMipMaps() () from /tmp/.mount_Olive-qMwbUl/usr/bin/../lib/libQt5Gui.so.5
#1  0x00007ffff4c48e04 in ?? () from /tmp/.mount_Olive-qMwbUl/usr/bin/../lib/libQt5Gui.so.5
#2  0x00007ffff4c4a489 in QOpenGLTexture::setData(int, QOpenGLTexture::PixelFormat, QOpenGLTexture::PixelType, void const*, QOpenGLPixelTransferOptions const*) () from /tmp/.mount_Olive-qMwbUl/usr/bin/../lib/libQt5Gui.so.5
#3  0x00000000004ca443 in ?? ()
#4  0x00000000004ef85d in ?? ()
#5  0x00000000004f0628 in ?? ()
#6  0x00000000004f1be0 in ?? ()
#7  0x00000000004f2075 in ?? ()
#8  0x00007ffff41fac5d in ?? () from /tmp/.mount_Olive-qMwbUl/usr/bin/../lib/libQt5Core.so.5
#9  0x00007ffff22dd6ba in start_thread (arg=0x7fffcf7fe700) at pthread_create.c:333
#10 0x00007ffff337741d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109

**Comment by Symbian9** [comment]
> Olive version: 789b751

Try latest https://github.com/olive-editor/olive/commit/13485a2587a5b947de25cdf26cdaac669625395d build

- https://github.com/olive-editor/olive/releases/tag/continuous

**Comment by AGI-THB** [comment]
Same Problem

[video](https://www.youtube.com/watch?v=wEfi_pwr7Zk)

**Comment by oc1024** [comment]
Full log with debug symbols:
```
2019-03-16T23:02:21 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:02:21 [WARNING] Tried to close an effect that was already closed
2019-03-16T23:02:21 [WARNING] Tried to start a closed effect - opening
2019-03-16T23:02:22 [WARNING] Tried to start a closed effect - opening
2019-03-16T23:02:22 [WARNING] QOpenGLTexturePrivate::destroy() called without a current context.
Texture has not been destroyed

Thread 16 "RenderThread" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffab79d700 (LWP 12253)]
0x00007ffff5a7cdb4 in QOpenGLTexture::setData(int, QOpenGLTexture::PixelFormat, QOpenGLTexture::PixelType, void const*, QOpenGLPixelTransferOptions const*) ()
   from /usr/lib/libQt5Gui.so.5
(gdb) bt
#0  0x00007ffff5a7cdb4 in QOpenGLTexture::setData(int, QOpenGLTexture::PixelFormat, QOpenGLTexture::PixelType, void const*, QOpenGLPixelTransferOptions const*)
    () at /usr/lib/libQt5Gui.so.5
#1  0x000055555565d24e in Effect::process_superimpose(double)
    (this=0x555556981640, timecode=1.6683350016683351)
    at effects/effect.cpp:909
#2  0x0000555555698ac3 in process_effect(Clip*, Effect*, double, GLTextureCoords&, unsigned int&, bool&, bool&, int)
    (c=0x555556a1b330, e=0x555556981640, timecode=1.6683350016683351, coords=..., composite_texture=@0x7fffab79c874: 0, fbo_switcher=@0x7fffab79c870: false, texture_failed=@0x7fffab79ca30: false, data=0)
    at rendering/renderfunctions.cpp:130
#3  0x0000555555699a4a in compose_sequence(ComposeSequenceParams&) (params=...)
    at rendering/renderfunctions.cpp:411
#4  0x000055555569bbec in RenderThread::paint() (this=0x7fffab79e010)
    at rendering/renderthread.cpp:181
#5  0x000055555569b814 in RenderThread::run() (this=0x7fffab79e010)
    at rendering/renderthread.cpp:112
#6  0x00007ffff4fcb96c in  () at /usr/lib/libQt5Core.so.5
#7  0x00007ffff4e60a9d in start_thread () at /usr/lib/libpthread.so.0
#8  0x00007ffff4a60b23 in clone () at /usr/lib/libc.so.6
(gdb)
```

In addition to that, a different backtrace generated from a similar crash.

```
2019-03-16T22:59:58 [WARNING] Tried to close an effect that was already closed
2019-03-16T22:59:58 [WARNING] Tried to start a closed effect - opening
2019-03-16T22:59:59 [WARNING] Tried to start a closed effect - opening
2019-03-16T22:59:59 [WARNING] Tried to start a closed effect - opening
2019-03-16T22:59:59 [WARNING] Tried to start a closed effect - opening

Thread 16 "RenderThread" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffab79d700 (LWP 12107)]
0x00007ffff5a77be0 in QOpenGLTexture::textureId() const ()
   from /usr/lib/libQt5Gui.so.5
(gdb) bt
#0  0x00007ffff5a77be0 in QOpenGLTexture::textureId() const ()
    at /usr/lib/libQt5Gui.so.5
#1  0x000055555565d25e in Effect::process_superimpose(double)
    (this=0x555556a26dc0, timecode=3.0363697030363697)
    at effects/effect.cpp:912
#2  0x0000555555698ac3 in process_effect(Clip*, Effect*, double, GLTextureCoords&, unsigned int&, bool&, bool&, int)
    (c=0x555556a20a30, e=0x555556a26dc0, timecode=3.0363697030363697, coords=..., composite_texture=@0x7fffab79c874: 0, fbo_switcher=@0x7fffab79c870: false, texture_failed=@0x7fffab79ca30: false, data=0)
    at rendering/renderfunctions.cpp:130
#3  0x0000555555699a4a in compose_sequence(ComposeSequenceParams&) (params=...)
    at rendering/renderfunctions.cpp:411
#4  0x000055555569bbec in RenderThread::paint() (this=0x7fffab79e010)
    at rendering/renderthread.cpp:181
#5  0x000055555569b814 in RenderThread::run() (this=0x7fffab79e010)
    at rendering/renderthread.cpp:112
#6  0x00007ffff4fcb96c in  () at /usr/lib/libQt5Core.so.5
#7  0x00007ffff4e60a9d in start_thread () at /usr/lib/libpthread.so.0
#8  0x00007ffff4a60b23 in clone () at /usr/lib/libc.so.6
(gdb) 
```

**Comment by Symbian9** [comment]
I can reproduce this bug with latest nightly build

*Olive* version: https://github.com/olive-editor/olive/commit/f0dd2ff82dc90cd20974bfb2c34fa154081647c6 (AppImage, x86_64)

**Comment by Symbian9** [comment]
Just tested latest https://github.com/olive-editor/olive/commit/bc6626e207ceb4ed0abc325bf6831bd60915d0b5 build — still has segfault.

BUT segfault reached if selected any clip with "Text" effect.

If unselect clips with "Text" effect then I could scroll timeline without segfault.

**Comment by itsmattkc** [comment]
Is this fixed in 67399702a0c73829cc734581517e2731e6609f83?

**Comment by AGI-THB** [comment]
Solved for me.
